MSBuild Logger Path are not correctly quoted

As a result, when a path with a space is used, the call to MSBuild does not work.

This is demonstrated here:

```sh
========================================
Build
========================================
Executing task: Build
Building src/Cake.DotNetCoreEf.sln
Executing: "C:/Program Files (x86)/Microsoft Visual Studio/2017/Professional/MSBuild/15.0/Bin/amd64/MSBuild.exe" /m /v:n
ormal /p:Configuration="Release" /p:Platform="Any CPU" /p:TreatWarningsAsErrors=true /target:Build /logger:XmlFileLogger
,C:/Visual Studio 2017/Projects/Cake.DotNetCoreEf/tools/MSBuild.Extension.Pack/tools/net40/MSBuild.ExtensionPack.Loggers
.dll;logfile="BuildArtifacts/MsBuild.log";invalidCharReplacement=_;verbosity=Detailed;encoding=UTF-8 "C:/Visual Studio 2
017/Projects/Cake.DotNetCoreEf/src/Cake.DotNetCoreEf.sln"
Microsoft (R) Build Engine version 15.1.548.43366
Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : error MSB1008: Only one project can be specified.
Switch: 2017/Projects/Cake.DotNetCoreEf/tools/MSBuild.Extension.Pack/tools/net40/MSBuild.ExtensionPack.Loggers.dll;logfi
le=BuildArtifacts/MsBuild.log;invalidCharReplacement=_;verbosity=Detailed;encoding=UTF-8

For switch syntax, type "MSBuild /help"
An error occurred when executing task 'Build'.
```

--- a/src/Cake.Common.Tests/Unit/Tools/MSBuild/MSBuildRunnerTests.cs
+++ b/src/Cake.Common.Tests/Unit/Tools/MSBuild/MSBuildRunnerTests.cs
@@ -846,7 +846,7 @@ namespace Cake.Common.Tests.Unit.Tools.MSBuild
                 var result = fixture.Run();
 
                 // Then
-                Assert.Equal("/v:normal /target:Build /logger:B,A;C /logger:E,D /logger:F " +
+                Assert.Equal("/v:normal /target:Build /logger:B,\"A\";C /logger:E,\"D\" /logger:\"F\" " +
                              "\"/Working/src/Solution.sln\"", result.Args);
             }
 

--- a/src/Cake.Common/Tools/MSBuild/MSBuildRunner.cs
+++ b/src/Cake.Common/Tools/MSBuild/MSBuildRunner.cs
@@ -6,6 +6,7 @@ using System;
 using System.Collections.Generic;
 using System.Globalization;
 using System.Linq;
+using System.Text;
 using Cake.Core;
 using Cake.Core.Diagnostics;
 using Cake.Core.IO;
@@ -126,10 +127,7 @@ namespace Cake.Common.Tools.MSBuild
             // Got any file loggers?
             if (settings.FileLoggers.Count > 0)
             {
-                var arguments = settings.FileLoggers.Select((logger, index) =>
-                {
-                    return GetLoggerArgument(index, logger);
-                });
+                var arguments = settings.FileLoggers.Select((logger, index) => GetLoggerArgument(index, logger));
 
                 foreach (var argument in arguments)
                 {
@@ -163,20 +161,21 @@ namespace Cake.Common.Tools.MSBuild
 
         private static string GetLoggerArgument(MSBuildLogger logger)
         {
-            string argument = "/logger:";
+            var argumentBuilder = new StringBuilder("/logger:");
             if (!string.IsNullOrWhiteSpace(logger.Class))
             {
-                argument += string.Format("{0},{1}", logger.Class, logger.Assembly);
+                argumentBuilder.Append(logger.Class);
+                argumentBuilder.Append(",");
             }
-            else
-            {
-                argument += logger.Assembly;
-            }
+
+            argumentBuilder.Append(logger.Assembly.Quote());
+
             if (!string.IsNullOrWhiteSpace(logger.Parameters))
             {
-                argument += string.Concat(";", logger.Parameters);
+                argumentBuilder.Append(";");
+                argumentBuilder.Append(logger.Parameters);
             }
-            return argument;
+            return argumentBuilder.ToString();
         }
 
         private static string GetPlatformName(PlatformTarget platform, bool isSolution)

